### DISCUSSION TITLE
echo executed commands

### DISCUSSION THREAD

**Discussion by dev72** [iapyeh/sshscript#3] [Ideas]
Hello, I really like sshscript.
I usually write fabric tasks but sshscript is simpler and very effective.

I'd like to auto echo executed commands on remote host, not their stdout or stderr.
I tried --verbose and --debug options but they don't work in this way.
Do you think it is possible to add something like an --echo option to print executed commands?

Thank you.
Gabriel.

**Comment by iapyeh** [comment]
Hi Gabriel. Thanks for your feedback. I agree that "print executed commands" is a must have feature.
From the version 1.1.2, the behavior of --verbose and --debug have changed. Now they can be used at the same time.
Below are examples:

file: a.spy
```
$.connect('timgwong@remotehost')
$ls -l
```

Case 1: run with --debug 
```
$ sshscript a.spy --debug
```
Output: the executed command ("ls -l") was printed  at line #3 of output
```
2022-08-10 20:45:12:[sshscript]run /home/iap/a.spy
2022-08-10 20:45:12:[sshscript]connecting timgwong@remotehost:22
2022-08-10 20:45:12:[sshscript$]ssh execute: ls -l
2022-08-10 20:45:13:[sshscript]224778858 closing timgwong@remotehost:22
```

Case 2: run with --verbose
```
$ sshscript a.spy --verbose
```
Output: data received by stderr and stdout are printed
```
🟩total 2
🟩drwxrwxr-x  3 timgwong timgwong 4096 Aug  1 21:58 care
🟩drwxrwxr-x. 8 timgwong timgwong  134 Aug  5 16:12 workspaceOfIap
```

Case 3: run with --debug --verbose
```
$ sshscript a.spy --debug --verbose
```
Output: content of both --debug and --verbose are printed
```
2022-08-10 20:45:33:[sshscript]run /home/iap/a.spy
2022-08-10 20:45:33:[sshscript]connecting timgwong@remotehost:22
2022-08-10 20:45:33:[sshscript$]ssh execute: ls -l
🟩total 2
🟩drwxrwxr-x  3 timgwong timgwong 4096 Aug  1 21:58 care
🟩drwxrwxr-x. 8 timgwong timgwong  134 Aug  5 16:12 workspaceOfIap
2022-08-10 20:45:34:[sshscript]224799974 closing timgwong@remotehost:22
```

Does it solve the problem?

**Comment by dev72** [comment]
Yes, it works, thank you!
